nodejs/week3/app.js

```diff
@@ -0,0 +1,50 @@
+import knex from "knex";
+const db = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "0.0.0.0",
+    port: Number(process.env.DB_PORT) || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "pass",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = db.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+      if (orderBy.length > 0) {
+      query = query.orderByRaw('??', [orderBy]);
+    }
```

Comment:
Can you explain how the above modification prevents the SQL injection?

Reply:
Thank you so much for your time to review my homework :) Have a lovely day :)